app/src/main/java/com/github/se/wanderpals/ui/screens/suggestion/SuggestionDetail.kt

```diff
@@ -289,6 +283,59 @@ fun SuggestionDetail(
                                 newCommentText = ""
                                 submitComment = true // Set the trigger for side effects
                               }
+                            }
+                          },
+                          modifier = Modifier.testTag("SendButton")) {
+                            if (!editingComment) {
+                              Icon(
+                                  Icons.AutoMirrored.Outlined.Send,
+                                  contentDescription =
+                                      "Send") // Change the icon to a send icon if adding a comment
+                            } else {
+                              if (newCommentText.isNotBlank()) {
+                                Icon(
+                                    Icons.Outlined.Create,
+                                    contentDescription =
+                                        "Edit") // Change the icon to an edit icon if editing a
+                                // comment
+                              } else {
+                                Icon(
+                                    Icons.Outlined.Clear,
+                                    contentDescription = "Cancel",
+                                ) // Change the icon to a clear icon if editing a comment and the
+                                // text is empty
+                              }
+                            }
+                          }
+                    },
+                    keyboardOptions = KeyboardOptions.Default.copy(imeAction = ImeAction.Done),
+                    keyboardActions =
+                        KeyboardActions(
+                            onDone = {
+                              if (editingComment) {
```

Comment:
You duplicated this line, (see Sonar cloud), you should use a function and pass it around instead of duplicating the code! This will make sonarcloud happy!
if (editingComment)...